compiler/AST/AggregateType.cpp

```diff
@@ -3113,7 +3113,7 @@ void AggregateType::addClassToHierarchy(std::set<AggregateType*>& localSeen) {
 
       auto ifcActuals = new CallExpr(PRIM_ACTUALS_LIST, new SymExpr(implementFor));
       auto istmt = ImplementsStmt::build(isym->name, ifcActuals, nullptr);
-      this->symbol->defPoint->insertAfter(istmt);
+      this->symbol->getModule()->block->insertAtTail(istmt);
```

Comment:
I would like to see an explanation of why this change is necessary.

Reply:
Updated the OP to include this:

>  While there, it fixes a bug in which nested aggregate types caused segmentation faults when an `: interface` syntax was used. This was because the `implements` statement for the child class was inserted into the body of the parent class, which only expected definitions (not a statement). Since flattening classes happens pretty early on, and moves child classes into the same scope as the parent class, the solution seemed to simply insert implements statements in module scope, like `FlattenClasses` does.